### PR TITLE
Completed spring05

### PR DESCRIPTION
**Pull request by Vedroid**

### CODE REVIEW

File: src/main/java/ua/vedroid/cinema/model/dto/MovieSessionRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package ua.vedroid.cinema.model.dto;
+
+public class MovieSessionRequestDto {
+    private Long id;
+    private String movieTitle;
```
Comment (Max-Pochepets) [inline src/main/java/ua/vedroid/cinema/model/dto/MovieSessionRequestDto.java:4]: Feel free to pass the id of the movie

File: src/main/java/ua/vedroid/cinema/model/dto/MovieSessionResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package ua.vedroid.cinema.model.dto;
+
+public class MovieSessionResponseDto {
+    private Long id;
+    private String movieTitle;
```
Comment (Max-Pochepets) [inline src/main/java/ua/vedroid/cinema/model/dto/MovieSessionResponseDto.java:5]: and return it

File: src/main/java/ua/vedroid/cinema/service/mapper/DtoMapper.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package ua.vedroid.cinema.service.mapper;
+
+public interface DtoMapper<E, Q, P> {
```
Comment (Max-Pochepets) [inline src/main/java/ua/vedroid/cinema/service/mapper/DtoMapper.java:3]: Interface Segregation. I actually don't think it is a good idea to make a single generic interface. Maybe we should do two interfaces, cause in the future we'll have situations, where a model won't need either ResponseDto, or RequestDto

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/ua/vedroid/cinema/service/mapper/DtoMapper.java (hunk 3 above)
Comment (boroda4436) [reply src/main/java/ua/vedroid/cinema/service/mapper/DtoMapper.java:3]: In this case, you can use generics like DtoMapper<Movie, MovieDto, MovieDto> @Max-Pochepets

File: src/main/java/ua/vedroid/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @PutMapping
+    public String update(@RequestBody MovieSessionRequestDto dto) {
+        MovieSession movieSession = service.update(mapper.toEntity(dto));
+        return "MovieSession " + movieSession + " - updated";
+    }
+
+    @DeleteMapping
+    public String remove(@RequestBody MovieSessionRequestDto dto) {
```
Comment (boroda4436) [inline src/main/java/ua/vedroid/cinema/controller/MovieSessionController.java]: please use id

File: src/main/java/ua/vedroid/cinema/model/dto/MovieSessionRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package ua.vedroid.cinema.model.dto;
+
+public class MovieSessionRequestDto {
+    private Long id;
```
Comment (boroda4436) [inline src/main/java/ua/vedroid/cinema/model/dto/MovieSessionRequestDto.java]: Do we need it?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/ua/vedroid/cinema/controller/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    @PutMapping("/{id}")
+    public String update(@PathVariable Long id, @RequestBody MovieSessionRequestDto dto) {
+        MovieSession movieSession = mapper.toEntity(dto);
+        movieSession.setId(id);
+        MovieSession updatedMovieSession = service.update(movieSession);
+        return "MovieSession " + updatedMovieSession + " - updated";
+    }
+
+    @DeleteMapping("/{id}")
+    public String remove(@PathVariable Long id, @RequestBody MovieSessionRequestDto dto) {
```
Comment (boroda4436) [inline src/main/java/ua/vedroid/cinema/controller/MovieSessionController.java]: do you still need a dto?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**